pkg/template/event.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2019 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package template
+
+import (
+	"regexp"
+	"strings"
+
+	pipelinev1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	"github.com/tidwall/gjson"
+	"golang.org/x/xerrors"
+)
+
+// eventPathVarRegex determines valid event path variables
+var eventPathVarRegex = regexp.MustCompile(`\$\(event(.[0-9A-Za-z_-]+)*\)`)
+
+// getEventPathFromVar returns the event path given an event path variable
+// $(event.my.path) -> my.path
+func getEventPathFromVar(eventPathVar string) string {
+	// Assume eventPathVar matches the eventPathVarRegex
+	if eventPathVar == "$(event)" {
+		return ""
+	}
+	return strings.TrimSuffix(strings.TrimPrefix(eventPathVar, "$(event."), ")")
+}
+
+// ApplyEventToParams returns the params with each event path variable replaced
+//  with the appropriate data from the event.
+// ApplyEventToParams returns an error when the event path variable is not found
+// in the event.
+func ApplyEventToParams(event []byte, params []pipelinev1.Param) ([]pipelinev1.Param, error) {
+	for i := range params {
+		param, err := applyEventToParam(event, params[i])
+		if err != nil {
+			return nil, err
+		}
+		params[i] = param
+	}
+	return params, nil
+}
+
+// applyEventToParam returns the param with each event path variable replaced
+// with the appropriate data from the event.
+// applyEventToParam returns an error when the event path variable is not found
+// in the event.
+func applyEventToParam(event []byte, param pipelinev1.Param) (pipelinev1.Param, error) {
+	// Get each event path variable in the param
+	eventPathVars := eventPathVarRegex.FindAllString(param.Value.StringVal, -1)
+	for _, eventPathVar := range eventPathVars {
+		eventPath := getEventPathFromVar(eventPathVar)
+		var eventValue string
+		if eventPath == "" {
```

Comment:
maybe a comment here? i dont quite see why an empty `eventPath` means that we want to treat the value of `event` differently

Reply:
I added a comment: `// $(event) has an empty eventPath, so use the entire event as the eventValue`